### PR TITLE
feat: add citation info

### LINKED ISSUE #221: Add citation file
Create a citation file based on github docs
https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files

### PR DESCRIPTION
**Pull request by piotrgramacki**
Closes #221

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - AdjacencyNeighbourhood
 - (CI) Changelog Enforcer
 - Utility plotting module based on Folium and Plotly
+- Citations information
```
Comment (RaczeQ) [inline CHANGELOG.md:25]: Citation**s** or Citation?

File: CITATION.cff
Comment (RaczeQ) [inline CITATION.cff:1]: We can add info about the `license`, `repository-code` (apart from the URL) and `keywords` (you can look into tags I have applied for the repo).
`Identifier` will be added after publication, maybe the same would be applied to the `abstract` field.

https://citation-file-format.github.io/cff-initializer-javascript/#/
https://github.com/citation-file-format/citation-file-format/blob/main/schema-guide.md

Comment (piotrgramacki) [reply CITATION.cff:1]: Ok, I added those

File: CHANGELOG.md (hunk 1 above)
Comment (piotrgramacki) [reply CHANGELOG.md]: Done

**Review by Calychas — APPROVED** [review]
LGTM Piotr!

**Review by simonusher — APPROVED** [review]

**Pull request merged**